src/Kaponata.Android/Adb/AdbClient.Shell.cs

```
@@ -0,0 +1,41 @@
+﻿// <copyright file="AdbClient.Shell.cs" company="Quamotion bv">
+// Copyright (c) Quamotion bv. All rights reserved.
+// </copyright>
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Kaponata.Android.Adb
+{
+    /// <summary>
+    /// Contains the <c>ADB</c> shell methods for the <see cref="AdbClient"/> class.
+    /// </summary>
+    public partial class AdbClient
+    {
+        /// <summary>
+        /// Connects to the sync service.
+        /// </summary>
+        /// <param name="device">
+        /// The device to which to connect.
+        /// </param>
+        /// <param name="shellCommand">
+        /// The shell command to execute.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// A <see cref="CancellationToken"/> which can be used to cancel the asynchronous operation.
+        /// </param>
+        /// <returns>
+        /// The output <see cref="ShellStream"/>.
+        /// </returns>
+        public virtual async Task<ShellStream> ExecuteRemoteShellCommandAsync(DeviceData device, string shellCommand, CancellationToken cancellationToken)
+        {
+            var protocol = await this.TryConnectToAdbAsync(cancellationToken, false).ConfigureAwait(false);
```

Do we know whether a client can be re-used once you've entered the "shell" mode? For example, could you do `ExecuteRemoteShellCommand` and then pull or push a file _using the same client_?

If not, should we perhaps have all the other methods throw an `InvalidOperationException` (😄) if you try to execute something while the client is still in "shell" mode?

Or overkill for now, and have an issue for it?